third_party/ruby/ruby.BUILD

```diff
@@ -4,6 +4,12 @@ load("@com_stripe_ruby_typer//third_party/ruby:build-ruby.bzl", "ruby")
 
 ruby(
     bundler = "@bundler_2_1_4//file",
+    linkopts = select({
+        "@com_stripe_ruby_typer//tools/config:darwin": [
+            "-mlinker-version=400",
+        ],
```

Comment:
I do not have a problem with this being committed as-is, but I am curious whether you tried to make the build work with the linker (if there is one) in LLVM 12.

Reply:
The build works on big sur without this, but fails in mac ci with it. This was the easiest compromise for getting a build that worked for both.

Reply:
Oh! I haven't tried using the `lld` that's present in the clang download. I'd be up for trying that as a follow up task, though.